### ISSUE TITLE
Media Upload Issues

### ISSUE THREAD

**Issue by gudenau** [tuskyapp/Tusky#995]
When I attempt to upload pictures from my cell phone to https://mastodon.technology they almost always fail. No error other than "media upload failed".

* * * *
- Tusky Version: 4.1
- Android Version: 9.0
- Android Device: Pixel XL
- Mastodon instance (if applicable): https://mastodon.technology

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by connyduck** [comment]
Yes, there seem to be problems with our upload code. I will add better error messages so we can finally nail them down.

**Comment by nevillepark** [comment]
I'm also experiencing this. 

_______________

- Tusky version: 4.1
- OS: LineageOS 15.1 (Android 8.1)
- Device: Moto G5 Plus
- Mastodon instance: social.coop

**Comment by feld** [comment]
> almost always fail

 Are the files that work PNG, GIF, or videos -- something that "send reduced" wouldn't touch?

**Comment by mihudevcom** [comment]
@gudenau Can you try the latest version? Seems to be working fine to me.

I've had a similar issue on other instances but it was because of the instance itself, not the app.

**Comment by mihudevcom** [comment]
@nevillepark Can you try the latest version?
@gudenau @nevillepark For both of you: when you experience trouble uploading images try the instance in your browser too. If the issue happens on both then it's a problem with the instance. If it happens only on the app then it might be a problem with the app.

**Comment by nevillepark** [comment]
@mihudevcom It only happens with the app.

**Comment by wjt** [comment]
On a Pixel 3 running Tusky 5.1, with an account on mastodon.social, some (but not all) image uploads fail with “The upload failed” after a second. It seems that it's only screenshots (PNG) which I have cropped with the Pixel's built-in screenshot editor: full-screen screenshots (also PNG), new photographs, and existing photographs (JPEG) succeed.

**Comment by freyja-wildes** [comment]
Hello there,
I encounter issues too. On a private instance (only a few accounts) image upload on toot creation is inconsistent.
Just now, I tried to upload twice it failed.
Looked around in options, wrote a toot. Then I tried again, with the same image, and it worked.

Tusky version: 12
Android version: 9
Android device: Asus Zenfone 5

**Comment by l1ving** [comment]
Can reproduce, Pixel 2, Android 11, Tusky 15.1

Video uploads fail 3/3 tries, photo uploads work but still randomly fail

**Comment by tokenwizard** [comment]
Same. Running version 15.1 on LineageOS. Also has similar media issues with AndStatus and Fedilab, so it may be related to the instance?

**Comment by l1ving** [comment]
I'm running glitch-soc on my instance, I doubt that's the issue.

**Comment by ariasuni** [comment]
I can upload perfectly well from the webapp but not from Tusky, for the two instances I tested. So there’s something going on between Tusky and Mastodon somewhere that makes much more flaky than it should.

I find it weird that the progress ring fills up, and then there’s a weird delay where nothing happens for a few seconds and often, with my fast connection, it’s where it fails.

**Comment by ariasuni** [comment]
Note that the issue is still here in Tusky 16.0 beta 2. I tried to use logcat to see what’s going on, but nothing is logged.

**Comment by connyduck** [comment]
Oh wait. We are using `/api/v1/media` to upload media, but there is `/api/v2/media` which is asynchronous (it does not wait for the media to be processed before returning) and should take way shorter. Fix coming asap, but I am not sure if we can get it into Tusky v16.

**Comment by connyduck** [comment]
Can you please test if this still occurs with the current Tusky nightly? @ariasuni

**Comment by tokenwizard** [comment]
> Can you please test if this still occurs with the current Tusky nightly? @ariasuni

I have not noticed this issue recently.

**Comment by ariasuni** [comment]
It’s way, way better! But when uploading four photos at once, some just get stuck with the round loading bar at 99% instead of being «removed» like before, but it’s hard to reproduce so…

**Comment by connyduck** [comment]
I made another change, please test it in the next release

**Comment by DonF22** [comment]
Hi, I am having the same issue, "upload failed" for  Image files.  
Android 10, one plus 5 phone/app , Tusky 17.
image upload Works fine on desktop, no go on app.   
@DonF@noagendasocial.com
anything i should try? 
  thank you.

**Comment by DonF22** [comment]
> I made another change, please test it in the next release

Hi, is next release 17 or higher than 17. Thanks

**Comment by connyduck** [comment]
@DonF22 Version 18

**Comment by chri2** [comment]
lineage_gts4lv-userdebug 11 RQ3A.211001.001
Tusky 18.0
mastodon on social.librem.one

Upload does not work. It stopped working on my device a while ago, but I can't say exactly when.

Is there a way to get/provide debugging information about why the upload is failing? I tried uploading pictures (screenshot, camera, edited - all the same).

**Comment by toniher** [comment]
I suspect some issues here may be related to #2770
